docs/proposals/multiple-schedulers.md

```diff
@@ -0,0 +1,139 @@
+
+    This conflict is complex to deal with in api-server and etcd. Our current solution is to let kubelet
+    to do the conflict check and if the conflict happens, effected pods would be put back to scheduler
+    and waiting to be scheduled again. Implementation details are in later sections.
+
+## Where to start: initial design
+
+We definitely want the multi-scheduler design to be a generic mechanism. The following lists the changes
+we want to make in the first step.
+
+- Add an annotation in pod template: `scheduler.alpha.kubernetes.io/scheduler-name`, this is used to
+separate pods between schedulers. `scheduler-name` should match one of the schedulers' `scheduler-name`
+(keep reading for more explanation)
+- Add a `scheduler-name` to each scheduler. It is done by hardcode or as command-line argument. The
+Kubernetes native scheduler (now `kube-scheduler` process) would have the name as `kube-scheduler`
+- The `scheduler-name` plays an important part in separating the pods between different schedulers
+which is done by carefully designing the condition/logic of how each scheduler selects unscheduled
+pods to add to its queue. To be specific, a scheduler can add a pod to its queue if and only if:
+    1. The pod has no host, **AND**
+    2. The `scheduler.alpha.kubernetes.io/scheduler-name` specified in the pod annotation matches
+    the scheduler name.
+
+        One exception is the default scheduler (with name `kube-scheduler`) who should be allowed
+        to add a none host pod to its queue if: a.) the pod DOES NOT contain annotation with key
+        `scheduler.alpha.kubernetes.io` **OR** b.) the pod contains annotation `scheduler.alpha.kubernetes.io/kube-scheduler`
+
+- Dealing with conflicts. All schedulers must use predicate functions that are at least as strict as
+the ones that Kubelet applies when deciding whether to accept a pod, otherwise Kubelet and scheduler
+may get into an infinite loop where Kubelet keeps rejecting a pod and scheduler keeps re-scheduling
+it back the same node. To make it easier for people who write new schedulers to obey this rule, we will
+create a library containing the predicates Kubelet uses. (See issue [#12744](https://github.com/kubernetes/kubernetes/issues/12744).)
+
+In summary, in the initial version of this multi-scheduler design, we will achieve the following:
+
+- If a pod has the annotation `scheduler.alpha.kubernetes.io/kube-scheduler` or the user does not explicitly
+sets this annotation in the template, it will be picked up by default scheduler
+- If the annotation is set and refers to a valid `scheduler-name`, it will be picked up by the scheduler of
+specified `scheduler-name`
+- If the annotation is set but refers to an invalid `scheduler-name`, the pod will not be picked by any scheduler.
+Since there is no admission controller or other thing alike to verify the annotation, there will be no error or
```

Comment:
@timothysc 

Does this mean delete sentence "Since there is no admission controller ..."?

Thanks for the review!